### ISSUE TITLE
Audio forwarding but keep on-phone audio on

### ISSUE THREAD

**Issue by LaplaceTor** [Genymobile/scrcpy#3875]
- [x] I have checked that a similar [feature request](https://github.com/Genymobile/scrcpy/issues?q=is%3Aopen+is%3Aissue+label%3A%22feature+request%22) does not already exist.

**Is your feature request related to a problem? Please describe.**

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

I try stream some mobile games via scrcpy, not only the audio forwarding  but also the audio on-phone I need, that I can connect my headphone to the phone and have a no delay gameplay with an audio-on streaming

**Comment by rom1v** [comment]
As soon as the "remote submix" aufio source is captured, audio is disabled on the device. I didn't find any way to configure this behavior.

If someone finds a way, I would be happy to add an option.

**Comment by LaplaceTor** [comment]
> As soon as the "remote submix" aufio source is captured, audio is disabled on the device. I didn't find any way to configure this behavior.
> 
> If someone finds a way, I would be happy to add an option.

How about an app install to phone with using the AudioPlaybackCapture API?
I use some other streaming apps in the past few years, they use this API but frame rate is worse than scrcpy so much.
In Android developer document, it told me that:
The MediaProjection API allows apps to acquire a MediaProjection token that gives them one time access to **capture screen contents or audio**

It's **or** not **and** so is it possible to capture audio only with an app and send it via adb?

**Comment by rom1v** [comment]
> How about an app install to phone

Installing an additional app would add a lot of complexity (this implies 2 parts on the server side to coordinate with the client). As long as I can avoid that, I don't want to add an app. (refs #3517)

>  with using the AudioPlaybackCapture API?

The AudioPlaybackCapture API is the API used by [sndcpy](https://github.com/rom1v/sndcpy). But it lets apps decide whether that can be captured or not, so in practice many apps are not captured. Refs [blogpost](https://blog.rom1v.com/2023/03/scrcpy-2-0-with-audio/#history).

**Comment by Gamusok** [comment]
Isn't it possible then to have audio in both the computer and the android device?

**Comment by bombshit** [comment]
@rom1v Combining this with sndcpy can achieve this situation, but there is a non-negligible delay in the sound. I hope developers can find a better method. Secondly, I am thinking about how to solve the problem of scrcpy's audio forwarding function failing when the microphone is turned on on the mobile phone. I hope I can get help.

**Comment by rom1v** [comment]
To investigate: #4380

**Comment by TFWol** [comment]
Wow, the same question sure has been asked a lot.

I recommend [pinning the Issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/pinning-an-issue-to-your-repository) as a known Issue. It _might_ cut down on some of the repeat questions.

Maybe reword it along the lines of "Playing from Device/Headphones and PC at the same time"

**Comment by jurassicjordan** [comment]
glad to know this is something still being worked on, any chance of seeing this feature in the near future?

**Comment by rom1v** [comment]
> any chance of seeing this feature in the near future?

Yes, now: #5102.

**Comment by sequencerr** [comment]
For some reason it works Windows but not on Debian 12

**Comment by rom1v** [comment]
Released in scrcpy 2.6 :rocket: 

https://github.com/Genymobile/scrcpy/blob/master/doc/audio.md#duplication
